### PR TITLE
Add name as prefix to mark group and scale’s name 

### LINKED ISSUE #819: Add name for a spec & use spec name as prefix for scale name.
Name should be a top-level property.  

This would be extremely important for composition.  (In fact, composition charts should automatically generate names for each subspace.)  

Currently scale name are just channel names, but with compositions - we can have multiple Xs and multiple Ys, etc.   Using each sub-spec’s name as prefix for scale would help distinguish different scale for the same channel of different subspecs.

### PR DESCRIPTION
**Pull request by kanitw**
See log for each commit for more details. 

Fixes #819

### CODE REVIEW

**Comment by domoritz** [comment]
Cannot merge.

**Comment by kanitw** [comment]
Stacking PR like Google does will avoid this problem.... so there's a trade-off between the two approach.

**Comment by kanitw** [comment]
Anyway, mergable now.

File: src/encoding.ts [hunk 1]
```diff
@@ -11,6 +11,16 @@ export function countRetinal(encoding: Encoding) {
   return count;
 }
 
+export function channels(encoding: Encoding) {
+  var arr = [];
+  CHANNELS.forEach(function(channel) {
```
Comment (domoritz) [inline src/encoding.ts]: Want to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter ?

**Pull request merged**